ResourceStatusSystem/Client/ResourceStatus.py

```diff
@@ -114,8 +155,148 @@ def __updateSECache( self ):
       return rawCache
     return S_OK( getCacheDictFromRawData( rawCache[ 'Value' ] ) )
 
+  def __updateCECache( self ):
+    """ Method used to update the ComputingElementCache.
+
+        It will try 5 times to contact the RSS before giving up
+    """
+
+    meta = { 'columns' : [ 'Name', 'StatusType', 'Status' ] }
+
+    for ti in range( 5 ):
+      rawCache = self.rssClient.selectStatusElement( 'Resource', 'Status',
+                                                     elementType = 'ComputingElement',
+                                                     meta = meta )
+      if rawCache['OK']:
+        break
+      self.log.warn( "Can't get CE status", rawCache['Message'] + "; trial %d" % ti )
+      sleep( math.pow( ti, 2 ) )
+      self.rssClient = ResourceStatusClient()
+
+    if not rawCache[ 'OK' ]:
+      return rawCache
+    return S_OK( getCacheDictFromRawData( rawCache[ 'Value' ] ) )
+
+
 ################################################################################
 
+  def __getRSSComputingElementStatus( self, elementName, statusType ):
+    """
+    Gets from the cache or the RSS the ComputingElements status. The cache is a
+    copy of the DB table. If it is not on the cache, most likely is not going
+    to be on the DB.
+
+    There is one exception: item just added to the CS, e.g. new ComputingElement.
+    The period between it is added to the DB and the changes are propagated
+    to the cache will be inconsistent, but not dangerous. Just wait <cacheLifeTime>
+    minutes.
+    """
+
+    cacheMatch = self.ceCache.match( elementName, statusType )
+
+    self.log.debug( '__getRSSComputingElementStatus' )
+    self.log.debug( cacheMatch )
+
+    return cacheMatch
+
+  def __getCSComputingElementStatus( self, elementName, statusType, default ):
```

Comment:
There is no CE status in the CS: there has never been one. So this method in practice does not make much sense. Also, the CS path "Resources/ComputingElements" does not exist